flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/table/DynamicKafkaDeserializationSchemaTest.java

```diff
@@ -0,0 +1,199 @@
+ */
+
+package org.apache.flink.streaming.connectors.kafka.table;
+
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.streaming.connectors.kafka.partitioner.FlinkKafkaPartitioner;
+import org.apache.flink.table.data.RowData;
+
+import org.apache.flink.shaded.guava31.com.google.common.collect.ImmutableList;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.header.internals.RecordHeaders;
+import org.apache.kafka.common.record.TimestampType;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.nio.ByteBuffer;
+import java.util.Collection;
+import java.util.Optional;
+
+/** test. */
+public class DynamicKafkaDeserializationSchemaTest {
+
+    @ParameterizedTest
+    @MethodSource("configProvider")
+    public void testdeserialize(TestSpec testSpec) {
+        // ensure we have the method.
+        DynamicKafkaRecordSerializationSchema.initializeMethod();
+        MockDeserializationSchema<RowData> keyDeserialization = null;
+        if (!testSpec.valueOnly) {
+            keyDeserialization = new MockDeserializationSchema<>(true, testSpec.providesHeaders);
+        }
+        MockDeserializationSchema<RowData> valueDeserialization =
+                new MockDeserializationSchema<>(false, testSpec.providesHeaders);
+
+        int[] keyProjection = {};
+        if (!testSpec.valueOnly) {
+            keyProjection = new int[] {0, 1};
+        }
+        int[] valueProjection = {2, 3};
+        TypeInformation producedTypeInfo = null;
+        DynamicKafkaDeserializationSchema dynamicKafkaRecordDeserializationSchema =
+                new DynamicKafkaDeserializationSchema(
+                        4,
+                        keyDeserialization,
+                        keyProjection,
+                        valueDeserialization,
+                        valueProjection,
+                        false,
+                        null,
+                        producedTypeInfo,
+                        false);
+        byte[] key = {};
+        if (!testSpec.valueOnly) {
+            key = new byte[] {0};
+        }
+        byte[] value = {0};
+        ConsumerRecord<byte[], byte[]> message =
+                new ConsumerRecord<>(
+                        "",
+                        0,
+                        1L,
+                        1L,
+                        TimestampType.NO_TIMESTAMP_TYPE,
+                        testSpec.valueOnly ? 0 : 1,
+                        1,
+                        key,
+                        value,
+                        new RecordHeaders(),
+                        Optional.empty());
+        try {
+            dynamicKafkaRecordDeserializationSchema.deserialize(message, null);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
```

Comment:
You can use https://junit.org/junit5/docs/5.0.1/api/org/junit/jupiter/api/Assertions.html#fail-java.lang.Throwable- `fail(cause)` to fail the junit test - rather than wrapping and rethrowing